libs/autofile/autofile.go

```diff
@@ -1,10 +1,13 @@
 package autofile
 
 import (
+	"fmt"
 	"os"
 	"sync"
 	"time"
 
+	"github.com/pkg/errors"
```

Comment:
we should use https://github.com/tendermint/tendermint/blob/master/libs/common/errors.go

Reply:
I started with that but it was annoyingly difficult to sth simple as just annotating an error with additional info (the formatted string). Maybe, I was just using it wrong.

Reply:
Is there a simple example somewhere? 

I want a sentinel error as above `PermissionsChangedErr` (such that the caller can check against it) and simply annotate it with more info. I can use `ErrorWrap` but I do not see a way to get the annotated content without the whole stack trace a la:
```
=== RUN   TestOpenAutoFilePerms
--- PASS: TestOpenAutoFilePerms (0.00s)
	autofile_test.go:85: --= Error =--
		Data: "file permissions changed"
		Msg Traces:
		    0  /Users/ismail/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:133 - [/var/folders/qf/7nq2274n50v5x9s2y4jcmfgm0000gn/T/permission_test853068086]: expected file permissions : -rw-------, got: -rwxr-xr-x
		Stack Trace:
		    0  /Users/ismail/go/src/github.com/tendermint/tendermint/libs/common/errors.go:14
		    1  /Users/ismail/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:133
		    2  /Users/ismail/go/src/github.com/tendermint/tendermint/libs/autofile/autofile.go:60
		    3  /Users/ismail/go/src/github.com/tendermint/tendermint/libs/autofile/autofile_test.go:82
		    4  /usr/local/Cellar/go/1.10.3/libexec/src/testing/testing.go:777
		    5  /usr/local/Cellar/go/1.10.3/libexec/src/runtime/asm_amd64.s:2362
		--= /Error =--
PASS
```
That's way to verbose for this simple case. Basically, we just want the first msg trace (as possible with `github.com/pkg/errors` via `errors.WithMessage`). `cmn.NewErrorWithData` doesn't provide sth. similar either. `NewError` can not keep the orig. error/Error. 

Am I missing sth. obvious here?

Reply:
I just got rid of `github.com/pkg/errors` and type aliased a string. I think that is the simplest way to handle this.